docs/src/resources/two-side-column.md

```diff
@@ -0,0 +1,923 @@
+  wire [2:0] _GEN_1 = 2'h1 == sel ? b : _GEN_0; // @[main.scala 16:16 24:8]
+  assign out = 2'h0 == sel ? a : _GEN_1; // @[main.scala 16:16 19:8]
+endmodule
+```
+</td>
+
+<td>
+
+```scala mdoc:silent
+class CaseStatementModule extends Module {
+  val a, b, c= IO(Input(UInt(3.W)))
+  val sel = IO(Input(UInt(2.W)))
+  val out = IO(Output(UInt(3.W)))
+  out := 0.U
+
+  switch (sel) {
+    is ("b00".U) {
+      out := a 
+    }
+    is ("b01".U) { 
+      out := b
+    }
+    is ("b10".U) {
+      out := c
+    }
+  }
+};
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementModule)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Case statement Enum Module 
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module CaseStatementModule(
+  input        clock,
+  input        reset,
+  input  [2:0] a,
+  input  [2:0] b,
+  input        sel,
+  output [2:0] out
+);
+  wire [2:0] _GEN_0 = sel ? b : 3'h0; // @[main.scala 22:16 30:8 20:9]
+  assign out = ~sel ? a : _GEN_0; // @[main.scala 22:16 25:8]
+endmodule
+```
+</td>
+<td>
+
+
+```scala mdoc:silent
+class CaseStatementEnumModule1 extends Module {
+  
+  object AluMux1Sel extends ChiselEnum {
+  val selectRS1, selectPC = Value
+  }
+  
+import AluMux1Sel._
+   val a, b = IO(Input(UInt(3.W)))
+    val sel = IO(Input(AluMux1Sel()))
+  val out = IO(Output(UInt(3.W)))
+    out := 0.U
+    
+  switch (sel) {
+  is (selectRS1) {
+   out := a
+  }
+  is (selectPC) {
+   out := b
+  }
+}
+  };
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementEnumModule1)
+```
+</td>
+         </tr>
+  <tr>
+<td>
+
+```
+module CaseStatementEnumModule2 (input clk);
+ 
+  typedef enum {INIT, IDLE, START, READY} t_state;
+    t_state state;
+    
+    
+ 
+    always @(posedge clk) begin
+        case (state)
+            IDLE    : state = START;
+            START   : state = READY;
+            READY   : state = IDLE ;
+            default : state = IDLE ;
+        endcase
+    end
+    
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class CaseStatementEnumModule2 extends Module {
+  
+  object AluMux1Sel extends ChiselEnum {
+   val INIT  = Value(0x03.U) 
+    val IDLE  = Value(0x13.U) 
+    val START = Value(0x17.U) 
+    val READY = Value(0x23.U) 
+  }
+   import AluMux1Sel._
+    val state = RegInit(INIT)
+  val nextState = IO(Output(AluMux1Sel()))
+  nextState := state
+
+  
+  switch (state) {
+  is (INIT) {
+   state := IDLE   
+  }
+  is (IDLE) {  
+   state := START
+  }
+  is (START) { 
+   state := READY
+  }
+ is (READY) { 
+   state := IDLE
+  }
+  }
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementEnumModule2)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# SystemVerilog Interfaces
```

Comment:
@debs-sifive to provide a pointer for a good link

Reply:
https://www.doulos.com/knowhow/systemverilog/systemverilog-tutorials/systemverilog-interfaces-tutorial/